### PR TITLE
Refman debug mode

### PR DESCRIPTION
**Pull request by SkySkimmer**
When COQ_DEBUG_REFMAN is defined in the environment, duplicate everything we send to coqtop to a file /tmp/coqdomainXXXX.v

This spams many files so not enabled by default.

Also disable the antlr version check when the environment variable is set as I couldn't figure out how to install the right version on my machine (it seems to work fine with 4.9.1 instead of 4.7.2).

The temporary file is put explicitly in /tmp instead of the automatic tmpdir as otherwise it goes in some /tmp/ subdirectory created by dune and deleted by dune. If we can disable the deletion that could be nicer.

### CODE REVIEW

**Comment by SkySkimmer** [comment]
>The temporary file is put explicitly in /tmp instead of the automatic tmpdir as otherwise it goes in some /tmp/ subdirectory created by dune and deleted by dune. If we can disable the deletion that could be nicer.

cc @coq/build-maintainers

File: doc/tools/coqrst/repl/coqtop.py [hunk 1]
```diff
@@ -54,17 +55,23 @@ def __init__(self, coqtop_bin=None, color=False, args=None) -> str:
...
         self.coqtop = None
+        self.debugfile = None
 
     def __enter__(self):
         if self.coqtop:
             raise ValueError("This module isn't re-entrant")
         self.coqtop = pexpect.spawn(self.coqtop_bin, args=self.args, echo=False, encoding="utf-8")
         # Disable delays (http://pexpect.readthedocs.io/en/stable/commonissues.html?highlight=delaybeforesend)
         self.coqtop.delaybeforesend = 0
+        if not (os.getenv ("COQ_DEBUG_REFMAN") is None):
```
Comment (jfehrle) [inline doc/tools/coqrst/repl/coqtop.py:66]:
```suggestion
        if os.getenv ("COQ_DEBUG_REFMAN"):
```

None is falsy.  Same thing in 3 other places.

**Review by jfehrle — COMMENTED** [review]
How about a comment somewhere in the code that describes what the setting does?

IIUC the generated file doesn't include the coqtop output.  Seems like it might be hard to relate the output to the associated Coq commands.  Not sure this is generally useful; it may have helped for a one-off issue.

File: doc/tools/coqrst/repl/coqtop.py (hunk 1 above)
Comment (cpitclaudel) [reply doc/tools/coqrst/repl/coqtop.py:66]: Alternatively, `is not None`; but not `not (… is None)`

**Comment by Alizter** [comment]
We can tell the rule running sphinx that we are producing a directory target for the debug files to live in. This will allow for dune to be aware of these files and we can keep them separated in their own directory. Let me know if you would like to set something like this up. I am assuming of course that these files are being generated in the rule in `doc/dune`.

**Comment by SkySkimmer** [comment]
> Seems like it might be hard to relate the output to the associated Coq commands. Not sure this is generally useful; it may have helped for a one-off issue.

I don't know what you think is hard.

I've used this for at least 2 issues so it's not a one-off.

**Comment by SkySkimmer** [comment]
@Alizter do we really need dune to be aware of debug files? I'd rather have a way to run in debug mode such that it doesn't delete the https://github.com/ocaml/dune/blob/main/src/dune_engine/dtemp.ml#L3

**Comment by Alizter** [comment]
@SkySkimmer With directory targets we wouldn't need to be aware of the exact files but just the directory that they appear in.

**Comment by SkySkimmer** [comment]
There's 1 file per rst file.
I would not call it convenient as the filenames are not distinctive (I don't know how to get the current rst name) but grepping for a given source line to find the right file is not hard either.

File: doc/tools/coqrst/notations/parsing.py [hunk 2]
```diff
@@ -14,12 +14,14 @@
...
 
+import os
+
 # Only print ANTLR version warning once
 checked_version = False
 check_version = Recognizer.checkVersion
 def checkVersion_once(*args, **kwargs):
     global checked_version
-    if not checked_version:
+    if not checked_version and os.getenv ("COQ_DEBUG_REFMAN"):
```
Comment (Zimmi48) [inline doc/tools/coqrst/notations/parsing.py:24]: Is this an unrelated change meant to remove the annoying ANTLR warning that we get at each build? In this case, wouldn't it make more sense to make `checked_version = True` if `COQ_DEBUG_REFMAN` is not defined, instead of repeatedly checking the value of this env var?

**Review by Zimmi48 — APPROVED** [review]
I don't mind this change and I don't have an opinion on whether these files need to be tracked by the build system. Currently, you are generating them in `/tmp`? In this case, this shouldn't matter.

**Comment by ppedrot** [comment]
What is this blocking on?

**Comment by ppedrot** [comment]
Nobody seems to oppose this PR so far, so in absence of further reaction I'll merge this PR soon.

**Comment by ppedrot** [comment]
@coqbot merge now
